### DISCUSSION TITLE
Support for Vial

### DISCUSSION THREAD

**Discussion by sommerper** [getreuer/qmk-keymap#25] [Q&A] [answered]
Hi!
I was wondering if you have tried to implement this for Vial (https://github.com/vial-kb/vial-qmk).

Essentially I'd want to have static homerow mods defined but be able to dynamically assign all other keys.
I've successfully used `#include "features/achordion.h"` in QMK before and can hardly live without it now but when I try to build a firmware in qmk-vial I get the following error:
```
make chocv:viahr                                                                                                                                                     ─╯
Making chocv with keymap viahr

avr-gcc (GCC) 5.4.0
Copyright (C) 2015 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

Size before:
   text    data     bss     dec     hex filename
      0   28088       0   28088    6db8 chocv_viahr.hex

Compiling: quantum/keymap_introspection.c                                                           [OK]
Compiling: quantum/via.c                                                                            [OK]
Linking: .build/chocv_viahr.elf                                                                     [ERRORS]
 |
 | .build/obj_chocv_viahr/quantum/qmk_settings.o (symbol from plugin): In function `qmk_settings_init':
 | (.text+0x0): multiple definition of `get_tapping_force_hold'
 | .build/obj_chocv_viahr/quantum/keymap_introspection.o (symbol from plugin):(.text+0x0): first defined here
 | collect2: error: ld returned 1 exit status
 |
make[1]: *** [builddefs/common_rules.mk:270: .build/chocv_viahr.elf] Error 1
Make finished with errors
make: *** [Makefile:414: chocv:viahr] Error 1
```

Would you have any hints as to why it would work in QMK and not Vial-QMK?
Thanks again for a nice project.

**Comment by sommerper** [comment]
I take that back. I just tried implementing this on a Corne and I didn't get any errors.
It looks like its conflicting on certain keyboards only so it's probably not an issue with achordion.
Merry xmas!

**Comment by sommerper** [comment, marked as answer]
Ah of course!

I had forgotten to set `QMK_SETTINGS = no` in `rules.mk`.

**Comment by getreuer** [comment]
I haven't yet taken a look at Vial, but definitely curious. Between the Corne vs. the first keyboard, there are rules and configs at the "keyboard" level that might be to blame—sometimes a keyboard maker turns on some questionable options. Glad you worked it out, and thanks for sharing the fix!
